### PR TITLE
feat: CP support Table expandIcon

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution

- #46889

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | feat: CP support Table expandIcon |
| 🇨🇳 Chinese | feat: CP support Table expandIcon |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/config-provider/context.ts [hunk 1]
```diff
@@ -66,6 +68,12 @@ export interface ComponentStyleConfig {
   style?: React.CSSProperties;
 }
 
+export interface TableConfig<T extends AnyObject = AnyObject> extends ComponentStyleConfig {
```
Comment (MadCcc) [inline components/config-provider/context.ts:71]: 这里的范型和 Table 的数据类型有关，但实际上 ConfigProvider 提供的配置不关心具体的组件数据，可以看看是不是可以把 API 简化一下。

Comment (li-jia-nan) [reply components/config-provider/context.ts]: done~ 改成直接用 `TableProps` 取里面的属性了

File: components/config-provider/context.ts [hunk 2]
```diff
@@ -66,6 +67,12 @@ export interface ComponentStyleConfig {
   style?: React.CSSProperties;
 }
 
+export interface TableConfig extends ComponentStyleConfig {
+  expandable?: {
+    expandIcon?: NonNullable<TableProps['expandable']>['expandIcon'];
```
Comment (MadCcc) [inline components/config-provider/context.ts:72]: 已经是 optional 的了还需要 NonNullable 么

Comment (li-jia-nan) [reply components/config-provider/context.ts:72]: 要把 undefined 排除掉，不然会报错呢

image

**Review by MadCcc — APPROVED** [review]

**Pull request merged**